packages/visualizations/addon/models/table.js

```diff
@@ -105,9 +106,9 @@ function hasAllColumns(request, columns) {
   //retrieve everything but dateTime from metadata.columns
   let columnFields  = arr(arr(columns).rejectBy('field', 'dateTime')).mapBy('field'),
       dimensions    = arr(get(request, 'dimensions')).mapBy('dimension.name'),
-      metrics       = arr(get(request, 'metrics')).mapBy('canonicalName'),
+      metrics       = get(request, 'metrics').map(metric => metric.toJSON()),
       requestFields = [ ...dimensions, ...metrics ];
 
   return requestFields.length === columnFields.length &&
-    requestFields.every(field => columnFields.includes(field));
+    requestFields.every(field => !!columnFields.find(column => isEqual(column, field)));
```

Comment:
Can we compare by `canonicalName` as we did with line charts?

Reply:
I looked into this, but this case is a little different since we're comparing columns that aren't always metrics.

Reply:
Wait a sec, that means the field property is an object when its a metric and a string when its a dimension??
That would be very confusing.

Reply:
Wait a sec, that means the field property is an object when its a metric and a string when its a dimension??
That would be very confusing.

Reply:
I could make dimensions and datetime columns store objects as well, but each object would have a different form so checking for the contents of the field object would still require a type check of the column. Maybe there should be a common `field` object form that all column types use